app/frontend/src/components/common/admin/inspection/InspectionPanel.vue

```diff
@@ -152,16 +152,24 @@
 import moment from 'moment';
 import { mapGetters } from 'vuex';
 
-import StatusTable from '@/components/minesoperatorscreening/admin/inspection/StatusTable.vue';
+import StatusTable from '@/components/common/admin/inspection/StatusTable.vue';
 import commonFormService from '@/services/commonFormService';
-import { AppClients, AppRoles, FormNames } from '@/utils/constants';
+import { AppRoles } from '@/utils/constants';
 
 export default {
   name: 'InspectionPanel',
   components: {
     StatusTable
   },
   props: {
+    formName: {
+      type: String,
+      required: true
+    },
+    resource: {
+      type: String,
+      required: true
+    },
```

Comment:
There's gotta be a way to get the resource name based on the form name...
It's a lot of extra code and bug-vectors to have to tell each component "you are the MINES form and also you are the MINES resource"

Reply:
I think Jeremy added those constants in, see what he thinks maybe.

Reply:
yeah. it's not great. 'resource' is just formName prefixed with 'comfort-'
so we could hard-code that. eg:
`
hasReviewer() {
      return this.hasResourceRoles(comfort-${this.formName}, [AppRoles.REVIEWER]);
    }
`

Reply:
There's two ways we can go about handling this at least in terms of a "component".
1) We have a pragmatic separation between what a "view" is vs what a "component" is -> a view knows by virtue of being a view what "flavor of form" it is and what its role in the entire application tree is. A component does not know what "flavor context" it is, it only knows to perform a certain set of tasks, given a set of parameters.
2) We let the "component" know how to handle itself based on the current path - in this case, a "view" only behaves as a structured container for the vue-router to know what to point to.

Either of those two options has its benefits and drawbacks. In option 1, you gain logical separation between the view and component "domains", but it incurs the cost of needing more props on the components in order to function. Option 2 follows the route of "self dependency", so the components will inherently be able to discern what they need to do based on where they appear at. The drawback of this approach is that we are crossing the "pure vue" ideal where a component should not care about what context it is in, only what it needs to do.

I believe we need to make a firm decision architecturally on which of these two options we want to pursue in earnest. As of right now, we have a mix of both of these approaches because we haven't made up our mind on what is the best way forwards. Maybe there isn't a "best" way for this, but we need to become consistent with our approach quickly before this gets out of hand.

Reply:
I'm just talking about a string concatenation or lookup so an extra identifier doesn't have to be passed in to each component, that's all.

Reply:
Should we propose that all components take in only formName as a prop -> inside the component, if it needs to know the resource, it'll call something like "formNameResource" which is inside the constants utils that just maps it out to the correct AppClient? For most of them it's just tacking on comfort-???, but we also do have a client solely named comfort (which is outside of a "form" to begin with).

Reply:
Yes

Reply:
ok. so i'll create the formNameResource map in the constants file and use that.

Reply:
Added the getAppClient utility function in commit d843a456e5ab246a835d40a73470378c5a45433d on your PR.